lib/routemaster/middleware/metrics.rb

```diff
@@ -0,0 +1,66 @@
+module Routemaster
+  module Middleware
+    class Metrics
+      INTERACTION_KEY = 'api_client'.freeze
+
+      attr_reader :client, :source_peer
+
+      def initialize(app, client: nil, source_peer: nil)
+        @app    = app
+        @client = client
+        @source_peer = source_peer
+      end
+
+      def call(request_env)
+        return @app.call(request_env) unless can_log?(request_env)
+
+        increment_req_count(request_tags(request_env))
+
+        record_latency(request_tags(request_env)) do
+          @app.call(request_env).on_complete do |response_env|
+            increment_response_count(response_tags(response_env))
+          end
+        end
+      end
+
+      def increment_req_count(tags)
```

Comment:
Did you intend these three methods to be part of the public API of the `Metrics` class? Seems like they should be private and `#call` should be the only public method.